applications/vanilla/models/class.categorymodel.php

```diff
@@ -3601,6 +3601,103 @@ private static function recalculateAggregateCounts() {
         self::instance()->clearCache();
     }
 
+    /**
+     * Recalculate  CountDiscussions and CountComments columns for category.
+     *
+     * @param int $categoryId Category ID to update
+     * @return void
+     */
+    public static function recalculateCategoryCounts(int $categoryId) {
+        $sql = Gdn::sql()
+            ->select('DiscussionID', 'count', 'DiscussionCount')
+            ->select('CountComments', 'sum', 'CountComments')
+            ->from('Discussion')
+            ->where('CategoryID', $categoryId);
+
+        $exclude = Gdn::config('Vanilla.Archive.Exclude');
```

Comment:
Isn't this chunk essentially a duplicate of `DiscussionModel::addArchiveWhere`?

Reply:
For the sake of minimizing the scope of this PR, I'm okay with this being dupe for now.